mlir/utils/jenkins/parameterSweeps.py

```diff
@@ -0,0 +1,394 @@
+        print(f"""Convolution returned intorrect result
+Output = {runnerOut}
+Errors = {runnerErrs.decode('utf-8')}""", file=sys.stderr)
+        return TestResult.FAIL
+    return TestResult.PASS
+
+def grouper(iterable, n):
+    it = iter(iterable)
+    while True:
+        chunk = tuple(itertools.islice(it, n))
+        if not chunk:
+            return
+        yield chunk
+
+async def dropGoodConfig(config: ConvConfiguration) -> Union[TestResult, ConvConfiguration]:
+    """Test the given `params`, returning the corresponding `config` on failure
+    and `None` on success or inapplicability"""
+    result = await testConfig(config)
+    if not QUIET:
+        print(f"{result.name}: {config!r}")
+    if result == TestResult.FAIL:
+        return config
+    return result
+
+IterType = TypeVar('IterType')
+async def sweepParameters(paramIter: Iterable[IterType],
+    toConfig: Callable[[IterType], MLIROnlyConfig]) -> Tuple[int, int, List[MLIROnlyConfig]]:
+
+    CONCURRENT_TESTS = 100
+    failingConfigs = []
+    passed = 0
+    invalid = 0
+    configs = (c for c in (toConfig(p) for p in paramIter))
+    for configs in grouper((dropGoodConfig(c) for c in configs), CONCURRENT_TESTS):
+        configsFuture = asyncio.gather(*configs)
+        try:
+            configsResults = await configsFuture
+        except Exception as e:
+            configsFuture.cancel()
+            raise e
+        for result in configsResults:
+            if result == TestResult.PASS:
+                passed = passed + 1
+            elif result == TestResult.INVALID:
+                invalid = invalid + 1
+            else:
+                failingConfigs.append(result)
+
+    return (passed, invalid, failingConfigs)
+
+CONV_STRUCTURE = itertools.product(
+    # Small/large
+    [False, True],
+    # op
+    ['fwd', 'wrw', 'bwd'],
+    # layout
+    ['NCHW', 'NHWC'],
+    # dtype
+    ['f32', 'f16'],
+    # Padding - hl, hr, wl, wr in [0, 3]
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    # Stride - 1 or 2
+    range(1, 3),
+    range(1, 3),
+    # Dilation in [1, 2]
+    range(1, 3),
+    range(1, 3))
+
+def to_conv_structure_type_test(params) -> MLIROnlyConfig:
+    size, op, layout, dtype, phl, phr, pwl, pwr, sh, sw, dh, dw = params
+    g, hi, wi, y, x = 1, 4, 4, 2, 2
+    if size:
+        n, c, k = 64, 64, 64
+    else:
+        n, c, k = 1, 7, 7
+    return MLIROnlyConfig(dtype, op, layout, n, c, hi, wi, k, y, x, sh, sw,
+        phl, phr, pwl, pwr, dh, dw, g, XDLOPS)
+
+XDLOPS_PERF_CONFIG = itertools.product(
+    # op
+    ['fwd', 'wrw', 'bwd'],
+    # layout
+    ['NCHW', 'NHWC'],
+    # dtype
+    ['f32', 'f16'],
+    # MPerBlock (exponent)
+    range(2, 9),
+    # NPerBlock (exponent)
+    range(4, 9),
+    # KPerBlock (exponent)
+    range(0, 4),
+    # MPerWave (exponent)
+    range(1, 8),
+    # NPerWave (exponent)
+    range(4, 8),
+    # KPack (exponent)
+    range(1, 4)
```

Comment:
[Note] I'm reminding myself that the ranges need to be adjusted when it is amended from MIOpen. How can we remember to synchronize those fields?